backends/tsdb/backend.go

```diff
@@ -110,15 +121,44 @@ func (b *Backend) newAdapter(session *frames.Session, password string, token str
 	return adapter, nil
 }
 
+// Get underlying bytes of string for read-only purposes to avoid allocating a slice.
+func getBytes(str string) []byte {
+	hdr := *(*reflect.StringHeader)(unsafe.Pointer(&str))
+	return *(*[]byte)(unsafe.Pointer(&reflect.SliceHeader{
+		Data: hdr.Data,
+		Len:  hdr.Len,
+		Cap:  hdr.Len,
+	}))
+}
+
 // GetAdapter returns an adapter
 func (b *Backend) GetAdapter(session *frames.Session, password string, token string, path string) (*tsdb.V3ioAdapter, error) {
-	// TODO: maintain adapter cache, for now create new per read/write request
 
-	adapter, err := b.newAdapter(session, password, token, path)
-	if err != nil {
-		return nil, err
+	h := fnv.New64()
+	_, _ = h.Write(getBytes(session.Url))
+	_, _ = h.Write(getBytes(session.Container))
+	_, _ = h.Write(getBytes(session.Path))
+	_, _ = h.Write(getBytes(session.User))
+	_, _ = h.Write(getBytes(session.Password))
```

Comment:
shouldn't you use `password, token and path` parameters instead of the session ones?